src/promptflow-devkit/promptflow/_sdk/operations/_trace_operations.py

```diff
@@ -81,3 +90,83 @@ def list_line_runs(
             line_run._append_evaluations(eval_line_runs)
             line_runs.append(line_run)
         return line_runs
+
+    @monitor_operation(activity_name="pf.traces.delete", activity_type=ActivityType.PUBLICAPI)
+    def delete(
+        self,
+        run: typing.Optional[str] = None,
+        collection: typing.Optional[str] = None,
+        started_before: typing.Optional[typing.Union[str, datetime.datetime]] = None,
+    ) -> None:
+        """Delete traces permanently.
+
+        Support delete according to:
+          - run
+          - non default collection
+          - collection combined with time as started before
+
+        Examples:
+          - pf.traces.delete(run="name")
+          - pf.traces.delete(collection="collection")
+          - pf.traces.delete(collection="default", started_before="2024-03-19T15:17:23.807563")
+
+        :param run: Name of the run.
+        :type run: Optional[str]
+        :param session: Id of the session.
+        :type session: Optional[str]
+        :param started_before: ISO 8601 format time string (e.g., "2024-03-19T15:17:23.807563").
+        :type started_before: Optional[Union[str, datetime.datetime]]
+        """
+        self._logger.debug(
+            "delete traces with parameters, run: %s, collection: %s, started_before: %s",
+            run,
+            collection,
+            started_before,
+        )
+        self._validate_delete_query_params(run=run, collection=collection, started_before=started_before)
+        self._logger.debug("try to delete line run(s)...")
+        if isinstance(started_before, str):
+            started_before = datetime.datetime.fromisoformat(started_before)
+        line_run_cnt, trace_ids = ORMLineRun.delete(
+            run=run,
+            collection=collection,
+            started_before=started_before,
+        )
+        self._logger.debug("deleted %d line runs", line_run_cnt)
+        self._logger.debug("try to delete traces and events for trace_ids: %s", trace_ids)
+        span_cnt = ORMSpan.delete(trace_ids=trace_ids)
+        event_cnt = ORMEvent.delete(trace_ids=trace_ids)
```

Comment:
maybe we should delete in reverse order? 

If somehow lineRun delete failed partially, we will lost the possibility to delete it's spans & event

Reply:
good catch, let me refine this within a transaction